GitHub Issue Forms: Simplify the bug report form template

priethor opened the pull request:
## Description
Followup to #33713. The first iteration on using form templates directly translated the markdown template to a form template. However, markdown templates allow for users to easily delete parts of it and adjust it to the report, whereas with the form template the report needs to be adapted entirely to the form. This leads to:
- When reporting, a long form that can't be tweaked and creates friction.
- Once reported, long reports even when optional fields are left empty

## Proposed changes
- Grouped the pre-checks into a single group.
- Moved the pre-checks to the bottom. Although being on top might help some users that are not aware of those recommendations, in general, it should be ok having them at the bottom similar to how checkboxes to "Accept terms" are at the bottom of forms. The main benefit, though, is avoiding having these checkboxes at the start of the bug report once submitted, where the most relevant information should be.
- Grouped together description and steps to reproduce.
- Grouped together expected vs actual behavior.
- Grouped together screenshots and code snippets as extra resources.
- Grouped together all environment-related fields (WordPress and Gutenberg versions, browser, device, OS).

Update after review feedback:
- Grouped together description, steps to reproduce, expected vs actual behavior.
- Make checkboxes optional

Update 2021-08-12:
- Split description and reproduction steps. 
- Removed "default theme" checkbox and moved the theme information to the environment section.

You can try the proposed changes [in this fork](https://github.com/priethor/gutenberg/issues/new?assignees=&labels=&template=Bug_report.yml&title=%3Ctitle%3E).

## Screenshots <!-- if applicable -->
Current bug report result if fields are empty:

![image](https://user-images.githubusercontent.com/27339341/129007407-f6e46fc7-7194-4006-8560-d18b7a0480c8.png)

Proposed new form (updated with the last changes after review):

![image](https://user-images.githubusercontent.com/27339341/129183823-808e5a5a-e346-49a6-9f51-0add7b8d173a.png)

priethor committed:
Simplify the bug report form template

priethor committed:
Fix yaml indentation

priethor committed:
Fix typo

ntsekouras commented on .github/ISSUE_TEMPLATE/Bug_report.yml:
```diff
@@ -52,76 +20,42 @@ body:
-              Please list what operating system you are using e.g. "Windows 10" or "iOS 14"
-      validations:
-          required: false
+              - label: I have searched the existing issues.
```
Should these remain required?

priethor replied:
Good call, I was thinking of them as "I accept the terms", but we probably don't need to enforce them, just remind them 👍 (like with the PR checklist)

ntsekouras reviewed (approved):
Thanks for working on this @priethor! I've created an issue today and the current form felt kind of overwhelming 😄 .

I like all the changes here! I personally I thought that `description` and `current behavior` sections could be combined, but your proposal works well IMO too.

annezazu commented on .github/ISSUE_TEMPLATE/Bug_report.yml:
```diff
@@ -6,42 +6,10 @@ body:
-          description: Please list the steps needed to reproduce the bug.
+          label: Description ands step-by-step reproduction instructions
+          description: Please write a brief description of the bug and the steps needed to reproduce it.
           placeholder: |
```
Since this is now for both a description and steps needed to reproduce, I'd include something about the description in the placeholder section.

priethor replied:
Any copy suggestions? I feel that adding "When I do '...', it happens '...' is way too similar to the Expected vs Current behavior.

annezazu commented on .github/ISSUE_TEMPLATE/Bug_report.yml:
```diff
@@ -52,76 +20,42 @@ body:
-          label: Expected Behavior
-          description: Please describe what you expected to happen.
+          label: Expected vs Current Behavior
+          description: Please describe what you expected to happen and what you are actually experiencing.
```
I'd change this to "Please describe what you expect to happen and what is currently happening."

priethor committed:
Make checkboxes optional

priethor committed:
Copy update

priethor commented:
Does anybody have strong feelings about combining description and expected vs current behavior? Saying what you expect and what actually happens feels like THE description to me. 😄  cc @annezazu @ntsekouras

ntsekouras commented:
>Saying what you expect and what actually happens feels like THE description to me

I agree 😄

priethor committed:
Merge description and expected/current behaviour in one field

priethor commented:
Updated the form to combine description, steps to reproduce, and expected vs actual behavior as part of the description, and updated the PR description accordingly.

priethor committed:
Copy update.

tellthemachines reviewed (commented):
Thanks, this is a great improvement! It makes total sense to combine description, expected and current behaviour because there's a lot of overlap in these fields. Steps to reproduce _could_ still be its own field, but perhaps we can revisit this if we find folks aren't adding them in properly.

We could also leave the pre-checks at the top of the form, as a reminder that these are the first steps to take in troubleshooting a bug. I'm not sure whether they should be required or not. Do we get a lot of reports that are actually plugin conflicts and the like? My feeling is no, but perhaps it's just because I haven't been doing a huge amount of triage 😅

talldan commented:
I have a slightly opposite opinion, and would like to keep the steps to reproduce separate from the description.

The steps to reproduce are the most important part of the bug report, it's crucial for devs to be able to test and debug the incoming reports. Often without it, reports are too vague to be able to reproduce and we have to ask for more information.

Most reporters are non-technical and don't quite understand what's important information and what isn't. I think combining them like this will lead to reporters only adding a description and skipping the steps to reproduce.

jasmussen commented:
Thanks for taking a stab. Can we revisit the question, "Have you tested with a default theme?". A few reasons:

- It's ambiguous for themes that exist in both classic and block versions.
- For older default themes like TwentyTwenty, they may have somewhat outdated editor styles that may suggest bugs in the editor that are in fact bugs in the theme itself. 

The reports are still valuable, regardless of theme, but just checking the box of testing with a default theme doesn't ensure the issue is that much more valid. We could just ask people to specify which theme they were testing with, that would be valuable.

javierarce commented:
> We could also leave the pre-checks at the top of the form, as a reminder that these are the first steps to take in 

I have to disagree with this and I'd like to suggest moving the pre-checks to a lower position (at the bottom of the form or at least under the description). 

Seeing the description right away is important. Otherwise, we'll need to scan the message and look for the description every time we land on an issue.

![image](https://user-images.githubusercontent.com/4933/129172320-4783f021-e1fd-4fa5-b04b-379f6c17f21c.png)


**Update**: Also, the original format for that list was better than the one above

![image](https://user-images.githubusercontent.com/4933/129183767-8fcacf82-5836-47ad-80d2-b2ed44e02ae0.png)

talldan commented:
> Seeing the description right away is important. Otherwise, we'll need to scan the message and look for the description every time we land on an issue.

If there was a way to have the checkbox visible on the form, but not in the published description, that would be ideal IMO. Probably not possible though.

javierarce commented:
> If there was a way to have the checkbox visible on the form, but not in the published description, that would be ideal IMO. Probably not possible though.

That's a good idea. Since issue forms are currently in beta, I'm going to write in their feedback forum:  https://github.com/github/feedback/discussions/categories/issues-feedback

priethor committed:
Separate steps field, move theme check to environment.

priethor commented:
Good feedback! I have taken the reproduction steps to a different field but left description, expected and actual behavior together, as they seem part of the description. I've also removed the default theme checked and added it to the environment data, it did feel the default theme check was too constrained by the reasons above:

![image](https://user-images.githubusercontent.com/27339341/129183995-363efd98-9b51-44ce-ad09-2454e128fa86.png)

priethor commented:
After letting the last changes sit for a day, I'm merging this for now. If more changes/feedback seem fit, or some of these don't improve the form, I'm happy to iterate over!

jasmussen commented:
👌

talldan commented:
This looks really good 👍 

Thanks for taking the suggestions into account!
